### PR TITLE
Add ability to cancel a submitted/running job from dashboard

### PR DESCRIPTION
**Pull request by cjkeilig**
This adds a cancel icon to each job in the job summary screen when it is running or idle. I added a function to open a modal, and another when the confirm button is pressed. I had to create a scope variable for the modal. I could not figure a way to send data to it directly. 

I did have to change the SlurmClusterInterface to account for jobs that are canceled before the automatic refreshCluster every 30 seconds. This was happening because jobs that aren't in the squeue are assumed to be complete in the jobStatusService... I'm trying to come up with a better way.

In reference to [#170](https://github.com/unlhcc/HCCGo/issues/170)

### CODE REVIEW

File: HCCGo/app/js/clusterLandingCtrl.js [hunk 1]
```diff
@@ -99,6 +100,24 @@ clusterLandingModule.controller('clusterLandingCtrl', ['$scope', '$log', '$timeo
...
   }
 
+  $scope.cancelModal = function(id, $event) {
+    $event.stopPropagation();
+    $scope.cancelNumber = id;
+    $("#cancel").modal("show");
+  }
+
+  $scope.cancelRunningJob = function(id, $event) {
+    var cancelledJob = $scope.jobs.find(function(x) { return x.jobId === id});
```
Comment (djw8605) [inline HCCGo/app/js/clusterLandingCtrl.js]: Why is this here?  Why can't you just pass the entire job object from above?

File: HCCGo/app/html/clusterLanding.html [hunk 2]
```diff
@@ -65,13 +84,18 @@ <h4 class="usage text-center">/work Usage</h4>
...
+              <i class="glyphicon glyphicon-check" ng-if="job.complete && !job.error && !job.cancelled" />
+              <i class="glyphicon glyphicon-ban-circle" ng-if="job.cancelled"/>
+              
             </div>
             <div class="row-content">
               <div class="action-secondary">
                 <a class="deleteButton" href="javascript:void(0)" ng-click="removeCompletedJob(job._id, $event)" ng-if="job.complete">
                   <i class="glyphicon glyphicon-remove"></i>
                 </a>
+                <a class="cancelButton" href="javascript:void(0)" ng-click="cancelModal(job.jobId, $event)"  ng-if="job.running || job.idle">
```
Comment (djw8605) [inline HCCGo/app/html/clusterLanding.html]: Pass the entire `job` object.

File: HCCGo/app/js/clusterLandingCtrl.js [hunk 3]
```diff
@@ -100,17 +100,37 @@ clusterLandingModule.controller('clusterLandingCtrl', ['$scope', '$log', '$timeo
...
+              "status": "COMPLETE",
+              "reportedStatus": "CANCELLED",
+              "justCancelled" : true
+            }
+          },
+          {},
+          function (err, numReplaced, affectedDocuments) {
+          });
+      });     
       $scope.refreshCluster(true);
```
Comment (djw8605) [inline HCCGo/app/js/clusterLandingCtrl.js]: I think your order of operations is off.  The refreshCluster should be done after the db is updated.

Comment (cjkeilig) [inline HCCGo/app/js/clusterLandingCtrl.js]: Would that mean putting the refresh inside the db update 'then' function?

Comment (djw8605) [inline HCCGo/app/js/clusterLandingCtrl.js]: Yes, in the then function.

**Comment by djw8605** [comment]
I don't see how the job is immediately updated to 'cancelled' state.  Immediately when the user confirms the cancel, the job should indicate it is cancelling some how.  Don't wait for scancel to return, and don't wait for updating the db.

**Comment by cjkeilig** [comment]
Ok, I can put the notification after the job booleans are set. I set these so that the user can't keep clicking the button and so that the job is marked as cancelled right away. Although, setting these and saving the complete status won't trigger an update for the job when the cluster is refreshed, so batch stats won't appear for the job.

File: HCCGo/app/js/clusterLandingCtrl.js [hunk 4]
```diff
@@ -99,6 +100,44 @@ clusterLandingModule.controller('clusterLandingCtrl', ['$scope', '$log', '$timeo
...
+  $scope.cancelModal = function(job, $event) {
+    $event.stopPropagation();
+    $scope.cancelJob = job;
+    $("#cancel").modal("show");
+  }
+
+  $scope.cancelRunningJob = function(job, $event) {
+    job.running = false;
+    job.idle = false;
+    job.cancelled = true;
```
Comment (djw8605) [inline HCCGo/app/js/clusterLandingCtrl.js]: Ah, I missed these changes.  This is what I was talking about.  This should work.

**Comment by djw8605** [comment]
Looks like we are still waiting on the few comments above.
